src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -193,7 +193,7 @@ public function getProgressPercent(): float
 
     public function getBarOffset(): int
     {
-        return floor($this->max ? $this->percent * $this->barWidth : (null === $this->redrawFreq ? min(5, $this->barWidth / 15) * $this->writeCount : $this->step) % $this->barWidth);
+        return floor($this->max ? $this->percent * $this->barWidth : (null === $this->redrawFreq ? (int) (min(5, $this->barWidth / 15) * $this->writeCount) : $this->step) % $this->barWidth);
```

Comment:
`intdic($this->barWidth, 15)`?

Reply:
Note that the result of the division is multiplied by `$this->writeCount`. We might lose precision if we cast too early.

Reply:
indeed, let's keep as is!